### PR TITLE
Add an option to ignore blank lines at the start/end of patterns and text

### PR DESCRIPTION
**Pull request by ltratt**
This is configurable, though I can't imagine many people wanting to turn this off!

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -224,14 +234,56 @@ impl<'a> FMatcher<'a> {
...
+                    if x.is_none() {
+                        return Ok(());
+                    }
+                    return Err(text_lines_off + skipped);
+                }
             }
         }
     }
 
+    /// If requested by the user, skip blank lines (including, optionally, the current blank line
```
Comment (vext01) [inline src/lib.rs]: First sentence here is quite long. I'd consider breaking it down.

**Comment by vext01** [comment]
Looks good. Just one meta-comment :)

File: src/lib.rs (hunk 1 above)
Comment (ltratt) [reply src/lib.rs]: This function is surprisingly complex, so it's a bit hard to document. Does the fix in https://github.com/softdevteam/fm/pull/4/commits/06411a4e309cd958738c377c0d818019968e44a6 work for you?

**Comment by vext01** [comment]
Much better, thanks.

Please squash.

**Comment by ltratt** [comment]
Squashed.

**Comment by ltratt** [comment]
@vext01 bors is stuck?

**Comment by vext01** [comment]
bors crashed.

bors r+

**Pull request merged**